### ISSUE TITLE
Addon-docs: Custom source snippet not rendering

### ISSUE THREAD

**Issue by JeffWeim** [storybookjs/storybook#11467]
**Describe the bug**
In the latest docs for `addon-docs` there is a [section](https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/recipes.md#customizing-source-snippets) describing how to customize the source code shown in the docs tab. However, when I try to use a custom snippet in a CSF file it does not seem to work (See code snippet below). 

**To Reproduce**
1. Create story using CSF
2. In export, attempt to override source snippet with:
  ```
  parameters: {
    docs: {
      source: {
        code: `custom snippet here`
      }
    }
  }
  ```
3. View storybook docs, the source snippet will be what is in the return statement, ie:
```javascript
  <Display element={elementValue} size={sizeValue}>{`Display ${sizeValue}`}</Display>
```

**Expected behavior**
I am expecting what is in the the `code: '<Display>Display Text Here</Display>'` string to be in the source snippet

**Screenshots**
n/a

**Code snippets**
Example CSF story:

```javascript
import React from 'react'
import { select } from '@storybook/addon-knobs'

import Display from './index'

const code = '<Display>Display Text Here</Display>'

export const base = () => {
  const sizeOptions = {
    xLarge: 'xLarge',
    large: 'large',
    medium: 'medium',
    small: 'small'
  }

  const elementOptions = {
    h1: 'h1',
    h2: 'h2',
    h3: 'h3',
    h4: 'h4'
  }

  const sizeValue = select('Size', sizeOptions, 'xLarge', 'size')
  const elementValue = select('Element', elementOptions, 'h1', 'element')

  return <Display element={elementValue} size={sizeValue}>{`Display ${sizeValue}`}</Display>
}

export default {
  component: Display,
  parameters: {
    docs: {
      source: {
        code
      }
    }
  }
}

```

**System:**
System:
    OS: macOS 10.15.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  Binaries:
    Node: 10.19.0 - ~/.nvm/versions/node/v10.19.0/bin/node
    Yarn: 1.22.4 - ~/.yarn/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v10.19.0/bin/npm
  Browsers:
    Chrome: 83.0.4103.116
    Firefox: 77.0.1
    Safari: 13.1
  npmPackages:
    @storybook/addon-a11y: ^6.0.0-beta.37 => 6.0.0-rc.0 
    @storybook/addon-docs: ^6.0.0-beta.37 => 6.0.0-rc.0 
    @storybook/addon-knobs: ^6.0.0-beta.37 => 6.0.0-rc.0 
    @storybook/react: ^6.0.0-beta.37 => 6.0.0-rc.0

**Additional context**
n/a

**Comment by JeffWeim** [comment]
@shilman This is seems to be resolved on the `6.0.0-rc.13` branch

**Comment by shilman** [comment]
Great! Thanks for letting me know @JeffWeim 👍
